diff --git a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-accordion/accordion.html.twig b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-accordion/accordion.html.twig
index b663324..cfcfe20 100644
--- a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-accordion/accordion.html.twig
+++ b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-accordion/accordion.html.twig
@@ -24,7 +24,7 @@
   {% set _classes = _classes|merge(['accordion-flush']) %}
 {% endif %}
 {% set attributes = attributes.addClass(_classes).setAttribute('id', 'accordion-' ~ _id) %}
-
+potatolopo
 {% if _items is not empty %}
 <div
   {{ attributes }}
diff --git a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-navbar/navbar.html.twig b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-navbar/navbar.html.twig
index 86db244..cef04c7 100644
--- a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-navbar/navbar.html.twig
+++ b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-navbar/navbar.html.twig
@@ -1,23 +1,23 @@
 {% spaceless %}
 
 {# Parameters:
-  - collapse_id (string) (default: '')
-  - background (string) (default: '')
-  - brand (object) (default: {})
-    format: {
-      src: '',
-      alt: '',
-      class: '',
-      label: '', (optional: only for string label)
-      link: '',
-    }
-  - navigation (object Navigation) (default: {})
-  - form (object Form) (default: {})
-  - expand (string) (default: 'lg')
-    options: ['sm', 'md', 'lg']
-  - color_set (string) (default: 'light')
-    options ['light', 'dark']
-  - disable_collapse (boolean) (default: false)
+- collapse_id (string) (default: '')
+- background (string) (default: '')
+- brand (object) (default: {})
+  format: {
+    src: '',
+    alt: '',
+    class: '',
+    label: '', (optional: only for string label)
+    link: '',
+  }
+- navigation (object Navigation) (default: {})
+- form (object Form) (default: {})
+- expand (string) (default: 'lg')
+  options: ['sm', 'md', 'lg']
+- color_set (string) (default: 'light')
+  options ['light', 'dark']
+- disable_collapse (boolean) (default: false)
 #}
 {% set _collapse_id = collapse_id|default('') %}
 {% set _background = background|default('') %}
@@ -45,25 +45,27 @@
   {{ attributes }}
 >
   <div class='container'>
-    {% if _brand is not empty %}
-      <a class='navbar-brand' href='{{ _brand.link }}'>
-      {%- if _brand.label is defined -%}
-        {{- _brand.label -}}
-      {%- else -%}
-        {% for logo in _brand.logos %}
-          <img
-            src='{{ logo.src }}'
-            {% if logo.alt is not empty %}
-            alt='{{ logo.alt }}'
-            {% endif %}
-            {% if logo.class is not empty %}
-            class='{{ logo.class }}'
-            {% endif %}
-          />
-        {% endfor %}
-      {%- endif -%}
-      </a>
-    {% endif %}
+    {% block navbar_branding %}
+      {% if _brand is not empty %}
+        <a class='navbar-brand' href='{{ _brand.link }}'>
+          {%- if _brand.label is defined -%}
+            {{- _brand.label -}}
+          {%- else -%}
+            {% for logo in _brand.logos %}
+              <img
+                src='{{ logo.src }}'
+                {% if logo.alt is not empty %}
+                  alt='{{ logo.alt }}'
+                {% endif %}
+                {% if logo.class is not empty %}
+                  class='{{ logo.class }}'
+                {% endif %}
+              />
+            {% endfor %}
+          {%- endif -%}
+        </a>
+      {% endif %}
+    {% endblock %}
     {% if not _disable_collapse %}
     <button
       class='navbar-toggler'
@@ -77,14 +79,18 @@
       <span class='navbar-toggler-icon'></span>
     </button>
     <div class='collapse navbar-collapse' id='{{ _collapse_id }}'>
-    {% endif %}
-      {% if _navigation is not empty %}
-        {% include '@oe-bcl/bcl-navigation/navigation.html.twig' with _navigation only %}
       {% endif %}
-      {% if _form is not empty %}
-        {% include '@oe-bcl/bcl-form/form.html.twig' with _form only %}
-      {% endif %}
-    {% if not _disable_collapse %}
+      {% block navbar_left %}
+        {% if _navigation is not empty %}
+          {% include '@oe-bcl/bcl-navigation/navigation.html.twig' with _navigation only %}
+        {% endif %}
+      {%  endblock %}
+      {% block navbar_right %}
+        {% if _form is not empty %}
+          {% include '@oe-bcl/bcl-form/form.html.twig' with _form only %}
+        {% endif %}
+      {%  endblock %}
+      {% if not _disable_collapse %}
     </div>
     {% endif %}
   </div>
