diff --git a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-carousel/carousel.html.twig b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-carousel/carousel.html.twig
index aa1b01a..8b96459 100644
--- a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-carousel/carousel.html.twig
+++ b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-carousel/carousel.html.twig
@@ -135,7 +135,7 @@
             <div class="{{ _item.caption_classes }}">
           {% endif %}
           {% if _item.caption_title is not empty %}
-            <p class="fs-5">{{ _item.caption_title }}</p>
+            <div class="fs-5">{{ _item.caption_title }}</div>
           {% endif %}
             {{- _item.caption -}}
           {% if _item.caption_classes is defined and _item.caption_classes is not empty %}
diff --git a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-date-block/date-block.html.twig b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-date-block/date-block.html.twig
index 9484d4a..e48d0c6 100644
--- a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-date-block/date-block.html.twig
+++ b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-date-block/date-block.html.twig
@@ -45,12 +45,12 @@
   <span class="d-flex flex-column pt-2">
     <span class="d-flex flex-column align-item-center justify-content-center bg-white border border-bottom-0 w-100 h-100 fs-5 text-uppercase">
       <span class="d-block">{{- _day ~ ' ' ~ _month ~ ((_end_day or _end_month) ? ' - ' ) -}}</span>
-      {%- if _end_day is not empty -%}
+      {%- if _end_day is not empty %}
         {{- _end_day -}}
-      {% endif %}
-      {% if _end_month is not empty %}
-        {{- _end_month -}}
-      {%- endif -%}
+      {% endif -%}
+      {%- if _end_month is not empty %}
+        {{- ' ' ~ _end_month -}}
+      {% endif -%}
     </span>
     <span class="bg-lighter border w-100 rounded-bottom py-1">
       {{- _year -}}
diff --git a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-gallery/gallery-item.html.twig b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-gallery/gallery-item.html.twig
index aeff199..962d5db 100644
--- a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-gallery/gallery-item.html.twig
+++ b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-gallery/gallery-item.html.twig
@@ -48,7 +48,7 @@ Parameters:
   {% if _item.caption is not empty or item.caption_title is not empty %}
     <div class="bcl-gallery__item-caption">
     {% if _item.caption_title is not empty %}
-      <p class="fs-5">{{ _item.caption_title }}</p>
+      <div class="fs-5">{{ _item.caption_title }}</div>
     {% endif %}
     {% if _item.caption is not empty %}
       <div class="bcl-gallery__item-description">
diff --git a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-link/link.html.twig b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-link/link.html.twig
index 615fcf3..aa94eb3 100644
--- a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-link/link.html.twig
+++ b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-link/link.html.twig
@@ -72,18 +72,19 @@
 {% if _classes is not empty %}
   {% set attributes = attributes.addClass(_classes) %}
 {% endif %}
-{% set attributes = attributes.setAttribute('href', _path) %}
 
-<a
-  {{ attributes }}
->
-  {%- if _icon_position == 'before' and _icon_markup is defined -%}
-    {{ _icon_markup|raw }}
-  {%- endif -%}
-  {{- _label -}}
-  {%- if _icon_position == 'after' and _icon_markup is defined -%}
-    {{ _icon_markup|raw }}
-  {%- endif -%}
-</a>
+{% if _icon_markup is defined %}
+  {%- set _label -%}
+    {%- if _icon_position == 'before' -%}
+      {{ _icon_markup }}
+    {%- endif -%}
+    {{- _label -}}
+    {%- if _icon_position == 'after' -%}
+      {{ _icon_markup }}
+    {%- endif -%}
+  {%- endset -%}
+{%- endif -%}
+
+{{ bcl_link(_label, _path, attributes) }}
 
 {% endapply %}
