diff --git a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-date-block/date-block.html.twig b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-date-block/date-block.html.twig
index 0fa903c..06f000c 100644
--- a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-date-block/date-block.html.twig
+++ b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-date-block/date-block.html.twig
@@ -46,12 +46,12 @@
   <span class="d-flex flex-column pt-2">
     <span class="d-flex flex-column align-item-center justify-content-center bg-white border border-bottom-0 w-100 h-100 fs-5 text-uppercase">
       <span class="d-block">{{- _day ~ ' ' ~ _month ~ ((_end_day or _end_month) ? ' - ' ) -}}</span>
-      {%- if _end_day is not empty -%}
+      {%- if _end_day is not empty %}
         {{- _end_day -}}
-      {% endif %}
-      {% if _end_month is not empty %}
-        {{- _end_month -}}
-      {%- endif -%}
+      {% endif -%}
+      {%- if _end_month is not empty %}
+        {{- ' ' ~ _end_month -}}
+      {% endif -%}
     </span>
     <span class="bg-lighter border w-100 rounded-bottom py-1">
       {{- _year -}}
diff --git a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-link/link.html.twig b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-link/link.html.twig
index 0ccc7ec..70b5143 100644
--- a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-link/link.html.twig
+++ b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-link/link.html.twig
@@ -1,4 +1,4 @@
-{% spaceless %}
+{% apply spaceless %}

 {# Parameters:
   - label (string) (default: '')
@@ -72,18 +72,20 @@
 {% if _classes is not empty %}
   {% set attributes = attributes.addClass(_classes) %}
 {% endif %}
-{% set attributes = attributes.setAttribute('href', _path) %}

-<a
-  {{ attributes }}
->
-  {%- if _icon_position == 'before' and _icon_markup is defined -%}
-    {{ _icon_markup|raw }}
-  {%- endif -%}
-  {{- _label -}}
-  {%- if _icon_position == 'after' and _icon_markup is defined -%}
-    {{ _icon_markup|raw }}
-  {%- endif -%}
-</a>
+{% if _icon_markup is defined %}
+  {%- set _label -%}
+    {%- if _icon_position == 'before' -%}
+      {{ _icon_markup }}
+    {%- endif -%}
+    {{- _label -}}
+    {%- if _icon_position == 'after' -%}
+      {{ _icon_markup }}
+    {%- endif -%}
+  {%- endset -%}
+{%- endif -%}
+
+{{ bcl_link(_label, _path, attributes) }}
+
+{% endapply %}

-{% endspaceless %}
