diff --git a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-carousel/carousel.html.twig b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-carousel/carousel.html.twig
index 6e08be9..ddc41ad 100644
--- a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-carousel/carousel.html.twig
+++ b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-carousel/carousel.html.twig
@@ -111,7 +111,7 @@
     {% if _item.image is defined and _item.image is not empty %}
       {{ _item.image }}
     {% endif %}
-    {% if (_item.caption is defined and _item.caption is not empty) or (_item.link is defined and _item.link is not empty) %}
+    {% if (_item.caption is defined and _item.caption is not empty) or (_item.link is defined and _item.link is not empty)  or (_item.caption_title is defined and _item.caption_title is not empty) %}
       {% set _caption_classes = 'carousel-caption' %}
       {% if _item.caption is not empty and _item.link is empty %}
         {% set _caption_classes = _caption_classes ~ ' d-none d-md-block' %}
diff --git a/node_modules/@openeuropa/bcl-theme-default/src/scss/base/_layout.scss b/node_modules/@openeuropa/bcl-theme-default/src/scss/base/_layout.scss
index 5d2dfcf..0e8a90f 100644
--- a/node_modules/@openeuropa/bcl-theme-default/src/scss/base/_layout.scss
+++ b/node_modules/@openeuropa/bcl-theme-default/src/scss/base/_layout.scss
@@ -3,3 +3,12 @@
     font-size: $h4-font-size;
   }
 }
+
+@include media-breakpoint-down(lg) {
+  .bcl-content-banner {
+    .bcl-card-start-col {
+      margin: 0 auto;
+      padding: 0 calc(var(--#{$variable-prefix}gutter-x) * 0.5);
+    }
+  }
+}
diff --git a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-base-templates/content-type.html.twig b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-base-templates/content-type.html.twig
index 96a5afb..5fe16ea 100644
--- a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-base-templates/content-type.html.twig
+++ b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-base-templates/content-type.html.twig
@@ -1,4 +1,4 @@
-{% spaceless %}
+{% apply spaceless %}
 
 {#
   Parameters:
@@ -106,7 +106,7 @@
       {% if _landing %}
       <div class="col-12">
       {% else %}
-      <div class="col-12 col-lg-10 col-xl-9 col-xxl-8">
+      <div class="col-lg-10 col-xl-9 col-xxl-8">
       {% endif %}
     {% endif %}
       {% block content_top %}
@@ -136,4 +136,4 @@
   {% include '@oe-bcl/bcl-footer/footer.html.twig' with footer only %}
 {% endif %}
 
-{% endspaceless %}
+{% endapply %}
diff --git a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-content-banner/content-banner.html.twig b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-content-banner/content-banner.html.twig
index b03e7b5..7bcc5fc 100644
--- a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-content-banner/content-banner.html.twig
+++ b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-content-banner/content-banner.html.twig
@@ -1,17 +1,17 @@
-{% spaceless %}
+{% apply spaceless %}
 
 {#
   Parameters:
-    - background: (string) (default: 'white')
-      options: [ white, gray ]
     - title: (string) (default: '')
     - title_tag: (string) (default: 'h5')
     - title_link: (link object) (default: {})
     - title_attributes: (drupal attrs)
+    - background: (string) (default: 'white')
+      options: [ white, gray ]
     - image (object)
     - image_size: (string)
     - date: (object)
-      options: ['sm', 'md', 'lg']
+      options: ['sm', 'md', 'lg', 'xl']
     - content: (string)
     - badges: (array of objects of type badge)
     - links: (array of objects of type link)
@@ -26,28 +26,28 @@
 {% set _title_link = title_link|default({}) %}
 {% set _title_attributes = title_attributes ?: create_attribute() %}
 {% set _background = background|default('white') %}
-{% set _action_bar = action_bar|default('') %}
 {% set _image = image|default({}) %}
-{% set _date = date|default({}) %}
 {% set _image_size = image_size|default('md') %}
+{% set _action_bar = action_bar|default('') %}
+{% set _date = date|default({}) %}
 
 {% if attributes is empty %}
   {% set attributes = create_attribute() %}
 {% endif %}
 
-{% set _css_classes = [
+{% set _classes = [
   'bcl-content-banner',
 ] %}
 
 {% if _background != 'white' %}
-  {% set _css_classes = _css_classes|merge([
+  {% set _classes = _classes|merge([
     'pb-md-4-75',
     'pb-4',
     'pt-4',
     'pt-md-4-75',
   ]) %}
 {% else %}
-  {% set _css_classes = _css_classes|merge([
+  {% set _classes = _classes|merge([
     'pt-4',
     'pt-md-4-75',
   ]) %}
@@ -59,24 +59,27 @@
 } %}
 
 {% set _body_classes = 'pt-0 px-0 pb-0' %}
-{% set _left_col_classes = '' %}
-{% set _right_col_classes = 'col-md-8 col-lg-9 col-xxl-8' %}
+{% set _right_col_classes = 'col-md col-lg-9 col-xxl-8' %}
 
 {% if _image is not empty %}
-  {% set _left_col_classes = 'col-8 col-md-4 col-lg-3 offset-2 offset-md-0' %}
-  {% set _right_col_classes = 'col-md-8 col-lg-9 col-xxl-8' %}
-  {% set _body_classes = 'pt-4 pt-md-0 px-0 px-md-3 pb-0' %}
+  {% set _left_col_classes = 'bcl-card-start-col' %}
+  {% set _body_classes = 'pt-4 pt-md-0 px-0 px-md-1-5 pb-0' %}
   {% if _image_size == 'lg' %}
-    {% set _left_col_classes = 'col-md-4' %}
-    {% set _right_col_classes = 'col-md-8' %}
+    {% set _left_col_classes = _left_col_classes ~ ' bcl-size-large' %}
+    {% set _right_col_classes = 'col-md col-xl-9 col-xxl-8' %}
   {% endif %}
-  {% if _image_size == 'sm' %}
-    {% set _left_col_classes = 'col-6 col-md-4 col-lg-2 offset-3 offset-md-0' %}
+  {% if _image_size == 'xl' %}
+    {% set _left_col_classes = _left_col_classes ~ ' bcl-size-extra-large' %}
+    {% set _body_classes = 'pt-4 pt-lg-0 px-0 px-lg-1-5 pb-0' %}
+    {% set _right_col_classes = 'col-lg' %}
   {% endif %}
 {% endif %}
 {% if _date is not empty %}
-  {% set _left_col_classes = 'col-8 col-md-3 col-lg-2 offset-4 offset-md-0' %}
-  {% set _right_col_classes = 'col-md-9 col-lg-10 col-xxl-8' %}
+  {% set _left_col_classes = 'bcl-card-start-col' %}
+  {% set _right_col_classes = 'col-12 col-md col-xxl-9' %}
+{% endif %}
+{% if _image is empty and _date is empty %}
+  {% set _right_col_classes = 'col-lg-10 col-xl-9 col-xxl-8' %}
 {% endif %}
 
 {% set _horizontal_classes = {
@@ -96,8 +99,8 @@
   attributes: create_attribute().addClass(['bg-transparent', 'border-0']),
 } %}
 
-{% set _css_classes = _css_classes|merge([_modifiers[_background]]) %}
-{% set attributes = attributes.addClass(_css_classes) %}
+{% set _classes = _classes|merge([_modifiers[_background]]) %}
+{% set attributes = attributes.addClass(_classes) %}
 
 {% if _button is not empty %}
   {% set _card =  _card|merge({
@@ -115,4 +118,4 @@
   </div>
 </div>
 
-{% endspaceless %}
+{% endapply %}
diff --git a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-pagination/pagination.html.twig b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-pagination/pagination.html.twig
index d33f381..ed77687 100644
--- a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-pagination/pagination.html.twig
+++ b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-pagination/pagination.html.twig
@@ -57,7 +57,7 @@
 {% set _alignment = alignment|default('') %}
 {% set _aria_label = aria_label|default('') %}
 {% set _enable_prev_next_icon = enable_prev_next_icon|default(false) %}
-{% set _enable_first_last_icon = enable_first_last_icon|default(true) %}
+{% set _enable_first_last_icon = enable_first_last_icon ?? true %}
 {% set _items = items|default([]) %}
 {% set _first = first|default({}) %}
 {% set _next = next|default({}) %}
