diff --git a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-link/link.html.twig b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-link/link.html.twig
index 171972f..aa94eb3 100644
--- a/node_modules/@openeuropa/bcl-theme-default/templates/bcl-link/link.html.twig
+++ b/node_modules/@openeuropa/bcl-theme-default/templates/bcl-link/link.html.twig
@@ -72,18 +72,19 @@
 {% if _classes is not empty %}
   {% set attributes = attributes.addClass(_classes) %}
 {% endif %}
-{% set attributes = attributes.setAttribute('href', _path) %}
 
-<a
-  {{ attributes }}
->
-  {%- if _icon_position == 'before' and _icon_markup is defined -%}
-    {{ _icon_markup }}
-  {%- endif -%}
-  {{- _label -}}
-  {%- if _icon_position == 'after' and _icon_markup is defined -%}
-    {{ _icon_markup }}
-  {%- endif -%}
-</a>
+{% if _icon_markup is defined %}
+  {%- set _label -%}
+    {%- if _icon_position == 'before' -%}
+      {{ _icon_markup }}
+    {%- endif -%}
+    {{- _label -}}
+    {%- if _icon_position == 'after' -%}
+      {{ _icon_markup }}
+    {%- endif -%}
+  {%- endset -%}
+{%- endif -%}
+
+{{ bcl_link(_label, _path, attributes) }}
 
 {% endapply %}
